core/unix/coredump.c

```diff
@@ -551,6 +554,10 @@ os_dump_core_internal(dcontext_t *dcontext)
         os_close(elf_file);
         return false;
     }
+    const char zero = 0;
+    for (int count = 0; count < padding; ++count) {
+        os_write(elf_file, &zero, sizeof(zero));
```

Comment:
One byte at a time seems rather inefficient. This is at most one page so we could use a local buffer and just one write system call, right?  (Or could try to do a seek or other method of advancing the file pointer.)

Reply:
I've added a buffer to call write() only once.